terraform/cloudwatch.tf.json.template.py

```diff
@@ -16,6 +22,21 @@
 )
 
 
+def lambda_resource_name(threshold: MetricThreshold) -> str:
+    if threshold.handler_name is None:
+        return threshold.lambda_name
+    else:
+        assert threshold.handler_name != ''
+        return '_'.join((threshold.lambda_name, threshold.handler_name))
```

Comment:
https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#collection-literals

And now you're over-using `join`. With two components that fit on one line, `+` is the obvious choice.

```suggestion
        return threshold.lambda_name + '_' + threshold.handler_name))
```